### DISCUSSION TITLE
Not able to add fish shell

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2614] [Getting started] [answered]
As I was told I ran several commands but at the end an error occurred that fish doesn't exist in that location(/usr/local/bin/fish). I have downloaded fish using homebrew.

**Comment by SMillerDev** [comment, marked as answer]
Are you on an M1 based mac? In that case it's `/opt/homebrew/bin/fish`

**Reply by ghost** [reply]
yes, so should I somehow change the /etc/shells to /opt/homebrew/bin/fish?

**Reply by carlocab** [reply]
```
echo /opt/homebrew/bin/fish | sudo tee --append /etc/shells
```

**Reply by ghost** [reply]
tee: illegal option -- -
usage: tee [-ai] [file ...]

**Reply by ghost** [reply]
I tried doing
 echo /opt/homebrew/bin/fish | sudo tee -a /etc/shells
/opt/homebrew/bin/fish

**Reply by ghost** [reply]
now my shells look like this
/opt/homebrew/bin/fish
/opt/homebrew/bin/fish

**Reply by carlocab** [reply]
It should look like this:
```
# List of acceptable shells for chpass(1).
# Ftpd will not allow users to connect who are not using
# one of these shells.

/bin/bash
/bin/csh
/bin/dash
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
/opt/homebrew/bin/fish
```

**Reply by ghost** [reply]
when I ran that command its now looking that way idk

**Reply by carlocab** [reply]
What is the output of `cat /etc/shells`?

**Reply by ghost** [reply]
/opt/homebrew/bin/fish
/opt/homebrew/bin/fish

**Reply by ghost** [reply]
@carlocab how do I make it look like that again?

**Reply by carlocab** [reply]
Do `sudo open /etc/shells` to open `/etc/shells` in TextEdit, then copy and paste the contents from above into the file and save it.

**Reply by ghost** [reply]
Its locked

**Reply by carlocab** [reply]
Hmmm, ok. Do
```
touch ~/shells
open -a TextEdit ~/shells
```
then paste the contents into the file that just opened, and save it. Do `cat ~/shells` to make sure it contains what you need it to have. Then do
```
sudo mv ~/shells /etc/shells
```

**Reply by ghost** [reply]
yep it did work let me try settings the shell now

**Reply by ghost** [reply]
@carlocab wait but how to set it as a shell in iTerm2 or default terminal?

**Reply by ghost** [reply]
here what it said when I made it default shell (chsh -s /opt/homebrew/bin/fish)

Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
fish: Unknown command: fish
~/.config/fish/functions/fish_prompt.fish (line 20): 
        fish --command "_tide_jobs_status=$status CMD_DURATION=$CMD_DURATION COLUMNS=$COLUMNS \
            fish_bind_mode=$fish_bind_mode set -U $_tide_prompt_var (_tide_prompt)" &
        ^
in function 'fish_prompt'
in command substitution
disown: There are no suitable jobs

**Reply by carlocab** [reply]
Is Homebrew in your `PATH`? You can do `echo $PATH` in `bash` or `zsh` to check. (That might also with with `fish`; not sure.)

**Reply by ghost** [reply]
/opt/homebrew/bin /opt/homebrew/sbin /usr/local/bin /usr/bin /bin /usr/sbin /sbin

**Reply by carlocab** [reply]
I'd try moving the contents of `~/.config/fish` elsewhere temporarily and see if you still get errors. If you don't, that means there's an issue with your config. If so, then this isn't a Homebrew issue, so you'll probably get better help elsewhere.

**Reply by ghost** [reply]
eh idk fish is kinda complex to setup, so I think ill stay on zsh with ohmyzsh
thx anyways!

**Comment by carlocab** [comment]
What's the output of `brew info fish`?

**Reply by ghost** [reply]
fish: stable 3.3.1 (bottled), HEAD
User-friendly command-line shell for UNIX-like operating systems
https://fishshell.com

**Reply by carlocab** [reply]
Is that all? It should be printing a lot more than that.
